zerver/views/registration.py

```diff
@@ -322,12 +327,19 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
             # make it respect invited_as_admin / is_realm_admin.
 
         if user_profile is None:
+            if request.POST.get('use_social_avatar') == 'on':
+                avatar_source = UserProfile.AVATAR_FROM_USER
```

Comment:
Wouldn't it be simpler to only pass `user_avatar_url` and `avatar_source = UserProfile.AVATAR_FROM_USER` in this ` request.POST.get('use_social_avatar') == 'on'` case? I don't think we should pass `user_avatar_url` other than `None` if it's not intended to be used. Perhaps `create_user` should even `assert not(avatar_source == UserProfile.AVATAR_FROM_GRAVATAR and user_avatar_url)` because that hypothetical combination of arguments doesn't really make sense.

Reply:
I'm not sure I understand what you saying. Like call `do_create_user` inside `if requests.POST.get('use_social_avatar') == 'on'` passing `avatar_source = UserProfile.AVATAR_FROM_USER` and `user_avatar_url`, otherwise call  `do_create_user` without these arguments? If it is, I think it makes the code on this function bigger than already is, no?

Reply:
An execution of what I meant would be to  perhaps change the block
```
    user_avatar_url = None
    if prereg_user.user_avatar_url:
        user_avatar_url = prereg_user.user_avatar_url
```
into
```
    user_avatar_url = None
    if prereg_user.user_avatar_url and  request.POST.get('use_social_avatar') == 'on':
        user_avatar_url = prereg_user.user_avatar_url
        avatar_source = UserProfile.AVATAR_FROM_USER
```
and then we can clean up the other block.

Reply:
aaaaaah ok